keras/engine/training.py

```diff
@@ -1332,6 +1334,11 @@ def _test_loop(self, f, ins, batch_size=None, verbose=0, steps=None):
             and/or metrics). The attribute `model.metrics_names` will give you
             the display labels for the scalar outputs.
         """
+        metrics_module.reset_global_metrics(self.metrics)
+        _, global_metric_names = metrics_module.get_global_metrics(self.metrics)
```

Comment:
Wouldn't : 
```
global_metric_indices = [self.metrics_names.index(name) for name in global_metrics_names]
```
Be more readable? An isn't name already a string so `str(name)` is redundant?

Reply:
I wanted a reusable helper function. ```str(name)```should be redundant but I'm always weary of string comparison in Python.

Reply:
Also worth noting, that it will not work with repeated metric names with that logic.